File: src/main/java/ProcessContextImpl.java
/**
 * @author Robin Duda
 *
 * wrapper for client services
 * should contain authentication info per process etc.
 */
public class ProcessContextImpl implements ProcessContext {

    public MyStupidPojo transform(MyStupidPojo pojo) {
        pojo.name = pojo.name + "_TRANSFORMED";
        System.out.println("TRANSFORMED");
        return pojo;
    }

    public void save(MyStupidPojo pojo) {
        System.out.println("saved pojo " + pojo.toString());
    }

}

File: src/main/kotlin/com/github/codingchili/model/ProcessContextImpl.kt
package com.github.codingchili.model

import com.github.codingchili.process.ProcessContext

/**
 * @author Robin Duda
 *
 * A very simple context, this is instantiated once per processor and reused during the
 * lifetime of the job, the context typically contains job configuration and helpful utilities.
 */
class ProcessContextImpl : ProcessContext {

    /**
     * Applies a simple transformation on the object being processed.
     */
    fun transform(pojo: ObjectForProcessing): ObjectForProcessing {
        pojo.name = pojo.name + "_TRANSFORMED"
        return pojo
    }

    /**
     * Performs a fake save operation on the element being processed.
     */
    fun save(pojo: ObjectForProcessing) {
        println("saved pojo $pojo")
    }
}
